### PR TITLE
removed TODOs from code

### PR DESCRIPTION
**Pull request by jameslamb**
In this PR, I propose removing all TODOs from our code and turning them into issues here on Github.

I would love to add a TODO-check to our CI but I'm not sure how to do that. You can edit the `script:` block in `.travis.yml` to configure what gets run, but I'm not sure what magic is happening to ensure that `R CMD CHECK` warnings result in build failures.

So for now, I just want to remove the existing TODOs and will open a separate issue for adding a CI step to prevent new ones from getting in.

### CODE REVIEW

File: R/AbstractGraphReporter.R [hunk 1]
```diff
@@ -441,7 +441,7 @@ AbstractGraphReporter <- R6::R6Class(
                 plotDTedges <- data.table::copy(self$edges) # Don't modify original
                 plotDTedges[, from := SOURCE]
                 plotDTedges[, to := TARGET]
-                plotDTedges[, color := '#848484'] # TODO Make edge formatting flexible too
+                plotDTedges[, color := '#848484']
```
Comment (jameslamb) [inline R/AbstractGraphReporter.R]: created #176 for this

File: R/AbstractGraphReporter.R [hunk 2]
```diff
@@ -482,10 +482,6 @@ AbstractGraphReporter <- R6::R6Class(
                 }
 
                 # Add legend
-                # but it is broken if there is only one level
-                # so hard-code for now to skip if there is only one level
-                # TODO: remove if statement when the following issue is resolved:
-                # https://github.com/datastorm-open/visNetwork/issues/290
```
Comment (jameslamb) [inline R/AbstractGraphReporter.R]: @jayqi I created #177, feel free to edit it if you think I didn't capture the intent of this

File: R/PackageDependencyReporter.R [hunk 3]
```diff
@@ -26,9 +26,6 @@ DependencyReporter <- R6::R6Class(
     "DependencyReporter",
     inherit = AbstractGraphReporter,
 
-    #TODO [patrick.boueri@uptake.com]: Add more robust error checks and logging
```
Comment (jameslamb) [inline R/PackageDependencyReporter.R]: this is legacy from an early version of `pkgnet`, the first time @Paxanator was like "hey classes are cool you should use classes".

File: R/PackageDependencyReporter.R [hunk 4]
```diff
@@ -26,9 +26,6 @@ DependencyReporter <- R6::R6Class(
...
-    #TODO [patrick.boueri@uptake.com]: Add version information to dependency structure
```
Comment (jameslamb) [inline R/PackageDependencyReporter.R]: created #178

File: R/PackageFunctionReporter.R [hunk 5]
```diff
@@ -210,8 +210,6 @@ FunctionReporter <- R6::R6Class(
             private$cache$edges <- private$extract_edges()
             log_info('Done extracting edges.')
 
-            # TODO (james.lamb@uptake.com):
-            # Make this handoff with coverage cleaner
```
Comment (jameslamb) [inline R/PackageFunctionReporter.R]: created #179

File: tests/testthat/test-DependencyReporter.R [hunk 6]
```diff
@@ -28,24 +28,24 @@ test_that('DependencyReporter structure is as expected', {
...
 ### USAGE OF PUBLIC AND PRIVATE METHODS AND FIELDS
 
 test_that('DependencyReporter Methods Work', {
     testObj <- DependencyReporter$new()
-    
+
     # inherited set_package
     expect_silent({
-        
+
```
Comment (jameslamb) [inline tests/testthat/test-DependencyReporter.R]: all these whitespace removals are from my default settings in RStudio. lmk if you want me to revert them, IMHO they're not a big deal

File: tests/testthat/test-FunctionReporter.R [hunk 7]
```diff
@@ -90,7 +90,6 @@ test_that('FunctionReporter Methods Work', {
                  , info = "derived edges do not match expected edges"
                  )
 
-    # TODO: Need to test that nodes were properly extracted
```
Comment (jameslamb) [inline tests/testthat/test-FunctionReporter.R]: created #180, @bburns632 feel free to edit the description on there to elaborate, I don't think what I've written so far is sufficient for a not-us developer to pick it up

File: tests/testthat/test-FunctionReporter.R [hunk 8]
```diff
@@ -102,13 +101,6 @@ test_that('FunctionReporter Methods Work', {
     expect_true(object = all(igraph::get.vertex.attribute(testPkgGraph)[[1]] %in% testNodeDT$node)
                 , info = "Graph nodes not as expected")
 
-    # Nodes table with coverage and metrics too
-    # TODO: Test that calculate_all_measures and other calculates attach metadata correctly
-    # expect_identical(object = sort(testObj$get_raw_data()$nodes$node)
-    #                  , expected = sort(testNodeDT$node)
-    #                  , info = "Different nodes than expected")
```
Comment (jameslamb) [inline tests/testthat/test-FunctionReporter.R]: @bburns632 this is covered by the tests below already, right?

File: tests/testthat/test-DependencyReporter.R (hunk 6 above)
Comment (bburns632) [reply tests/testthat/test-DependencyReporter.R]: There are all whitespace additions you are proposing, right?

File: tests/testthat/test-FunctionReporter.R (hunk 8 above)
Comment (bburns632) [reply tests/testthat/test-FunctionReporter.R]: yes and more so by https://github.com/UptakeOpenSource/pkgnet/blob/master/tests/testthat/test-NetworkMeasures.R

File: tests/testthat/test-FunctionReporter.R (hunk 7 above)
Comment (bburns632) [reply tests/testthat/test-FunctionReporter.R]: Should also be covered by https://github.com/UptakeOpenSource/pkgnet/blob/master/tests/testthat/test-NetworkMeasures.R

File: tests/testthat/test-DependencyReporter.R (hunk 6 above)
Comment (jameslamb) [reply tests/testthat/test-DependencyReporter.R]: no whitespace removals

File: tests/testthat/test-FunctionReporter.R (hunk 8 above)
Comment (jameslamb) [reply tests/testthat/test-FunctionReporter.R]: awesome, thank you

**Pull request merged**